Keycloak.X cannot lookup embedded theme-resources from extension jars

### Describe the bug

The Wildfly based Keycloak distribution supports accessing embedded templates from `theme-resources` folder in extension jars. 

The Keycloak.X distribution fails to resolve templates form `theme-resources`. 
During  boostrap the `ProviderManager` is configures a `KeycloakDeploymentInfo` with [service lookups enabled but without `themes-resource` support](https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java#L438). Therefore no `ClasspathThemeResourceProviderFactory` is configured and therefore nested theme-resource lookups are not supported.

In the Wildfly based Keycloak distribution the `ClasspathThemeResourceProviderFactory` is registered via the [DefaultProviderLoader](https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/services/src/main/java/org/keycloak/provider/DefaultProviderLoader.java#L65)

### Version

Keycloak.X 16.1.0

### Expected behavior

Keycloak.X should support nested theme-resource lookup as the Wildfly based Keycloak distribution does.

### Actual behavior

Nested theme-resources lookups do not work.

### How to Reproduce?

1) Create an extension (authenticator) with embedded resources
2) Deploy the extension with Keycloak.X.
3) Try to access the authenticator
4) The server console will contain a strack-trace indicating that the theme template could not be found.

### Anything else?

A naive fix could be to change the line:
https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java#L438

to
```java
ProviderManager pm = new ProviderManager(KeycloakDeploymentInfo.create().services().themes().themeResources(), classLoader);
```


As a workaround we explicitly register the `ClasspathThemeResourceProviderFactory` with a custom implementation:
```java
package com.github.thomasdarimont.keycloakx.themes;

import com.google.auto.service.AutoService;
import org.keycloak.theme.ClasspathThemeResourceProviderFactory;
import org.keycloak.theme.ThemeResourceProviderFactory;

/**
 * Workaround for missing {@link ClasspathThemeResourceProviderFactory} in Keycloak.X to support theme-resources in extension jars.
 */
@AutoService(ThemeResourceProviderFactory.class)
public class CustomClasspathThemeResourceProviderFactory extends ClasspathThemeResourceProviderFactory {

    private static final String ID = "custom-classpath-themeresource-provider";

    public CustomClasspathThemeResourceProviderFactory() {
        super(ID, Thread.currentThread().getContextClassLoader());
    }

    @Override
    public String getId() {
        return ID;
    }
}

```

@pedroigor unfortunately it's not so easy...

Using
```java
ProviderManager pm = new ProviderManager(KeycloakDeploymentInfo.create().services().themes().themeResources(), classLoader);
```
in `KeycloakProcessor` yields to an NPE at built-time...

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:48 min
[INFO] Finished at: 2022-01-20T20:06:44+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:2.5.4.Final:build (default) on project keycloak-quarkus-server-app: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.steps.MainClassBuildStep#build threw an exception: java.lang.RuntimeException: Failed to record call to method public void org.keycloak.quarkus.runtime.KeycloakRecorder.configSessionFactory(java.util.Map,java.util.Map,java.util.Map,java.lang.Boolean)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.writeBytecode(BytecodeRecorderImpl.java:505)
[ERROR] 	at io.quarkus.deployment.steps.MainClassBuildStep.writeRecordedBytecode(MainClassBuildStep.java:463)
[ERROR] 	at io.quarkus.deployment.steps.MainClassBuildStep.build(MainClassBuildStep.java:274)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:887)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.NullPointerException
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadComplexObject(BytecodeRecorderImpl.java:1113)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadObjectInstanceImpl(BytecodeRecorderImpl.java:934)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadObjectInstance(BytecodeRecorderImpl.java:606)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadComplexObject(BytecodeRecorderImpl.java:1115)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadObjectInstanceImpl(BytecodeRecorderImpl.java:934)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadObjectInstance(BytecodeRecorderImpl.java:606)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadComplexObject(BytecodeRecorderImpl.java:1115)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadObjectInstanceImpl(BytecodeRecorderImpl.java:934)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.loadObjectInstance(BytecodeRecorderImpl.java:606)
[ERROR] 	at io.quarkus.deployment.recording.BytecodeRecorderImpl.writeBytecode(BytecodeRecorderImpl.java:500)
[ERROR] 	... 13 more
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-quarkus-server-app
```

Any ideas with this could be?

--- a/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
+++ b/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
@@ -421,7 +421,14 @@ class KeycloakProcessor {
             Map<String, ProviderFactory> preConfiguredProviders) {
         Config.init(new MicroProfileConfigProvider());
         ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-        ProviderManager pm = new ProviderManager(KeycloakDeploymentInfo.create().services(), classLoader);
+
+        KeycloakDeploymentInfo keycloakDeploymentInfo = KeycloakDeploymentInfo.create()
+                .name("classpath")
+                .services()
+                // .themes() // handling of .jar based themes is already supported by Keycloak.x
+                .themeResources();
+
+        ProviderManager pm = new ProviderManager(keycloakDeploymentInfo, classLoader);
         Map<Spi, Map<Class<? extends Provider>, Map<String, ProviderFactory>>> factories = new HashMap<>();
 
         for (Spi spi : pm.loadSpis()) {

--- a/server-spi-private/src/main/java/org/keycloak/provider/KeycloakDeploymentInfo.java
+++ b/server-spi-private/src/main/java/org/keycloak/provider/KeycloakDeploymentInfo.java
@@ -37,6 +37,10 @@ public class KeycloakDeploymentInfo {
         return name;
     }
 
+    /**
+     * Enables discovery of services via {@link java.util.ServiceLoader}.
+     * @return
+     */
     public KeycloakDeploymentInfo services() {
         this.services = true;
         return this;
@@ -46,6 +50,10 @@ public class KeycloakDeploymentInfo {
         return themes;
     }
 
+    /**
+     * Enables discovery embedded themes.
+     * @return
+     */
     public KeycloakDeploymentInfo themes() {
         this.themes = true;
         return this;
@@ -55,6 +63,10 @@ public class KeycloakDeploymentInfo {
         return themeResources;
     }
 
+    /**
+     * Enables discovery of embedded theme-resources.
+     * @return
+     */
     public KeycloakDeploymentInfo themeResources() {
         themeResources = true;
         return this;

--- a/services/src/main/java/org/keycloak/theme/ClasspathThemeResourceProviderFactory.java
+++ b/services/src/main/java/org/keycloak/theme/ClasspathThemeResourceProviderFactory.java
@@ -22,6 +22,10 @@ public class ClasspathThemeResourceProviderFactory implements ThemeResourceProvi
     private final String id;
     private final ClassLoader classLoader;
 
+    public ClasspathThemeResourceProviderFactory() {
+        this("classpath", Thread.currentThread().getContextClassLoader());
+    }
+
     public ClasspathThemeResourceProviderFactory(String id, ClassLoader classLoader) {
         this.id = id;
         this.classLoader = classLoader;

